pkg/apis/alerting/v1/extensions.go

```diff
@@ -1,30 +1,71 @@
 
 import (
+	"fmt"
 	"time"
 
+	"github.com/google/uuid"
+	"github.com/prometheus/alertmanager/pkg/labels"
 	"github.com/rancher/opni/pkg/alerting/shared"
 	corev1 "github.com/rancher/opni/pkg/apis/core/v1"
+	"github.com/rancher/opni/pkg/capabilities/wellknown"
 	"github.com/samber/lo"
+	"golang.org/x/exp/slices"
 	"google.golang.org/protobuf/types/known/durationpb"
 	timestamppb "google.golang.org/protobuf/types/known/timestamppb"
 )
 
 const UpstreamClusterId = "UPSTREAM_CLUSTER_ID"
 const EndpointTagNotifications = "notifications"
 
+// Note these properties have to conform to the AlertManager label naming convention
+// https://prometheus.io/docs/alerting/latest/configuration/#labelname
 const (
-	// Any messages already in the queue with the same dedupe key will not be processed
-	NotificationPropertyDedupeKey = "opni.io/dedupeKey"
+	// maps to a wellknown.Capability
+	RoutingPropertyDatasource = "opni_datasource"
+)
+
+var (
+	OpniSubRoutingTreeMatcher *labels.Matcher = &labels.Matcher{
+		Type:  labels.MatchEqual,
+		Name:  RoutingPropertyDatasource,
+		Value: "",
+	}
+
+	OpniMetricsSubRoutingTreeMatcher *labels.Matcher = &labels.Matcher{
+		Type:  labels.MatchEqual,
+		Name:  RoutingPropertyDatasource,
+		Value: wellknown.CapabilityMetrics,
+	}
+
+	OpniSeverityTreeMatcher *labels.Matcher = &labels.Matcher{
+		Type:  labels.MatchNotEqual,
+		Name:  NotificationPropertySeverity,
+		Value: "",
+	}
+)
+
+// Note these properties have to conform to the AlertManager label naming convention
+// https://prometheus.io/docs/alerting/latest/configuration/#labelname
+const (
+	// Property that specifies the unique identifier for the notification
+	// Corresponds to condition id for `AlertCondition` and an opaque identifier for
+	// an each ephemeral `Notification` instance
+	NotificationPropertyOpniUuid = "opni_uuid"
+	// Any messages already in the notification queue with the same dedupe key will not be processed
+	// immediately, but will be deduplicated.
+	NotificationPropertyDedupeKey = "opni_dedupe_key"
 	// Any messages with the same group key will be sent together
-	NotificationPropertyGroupKey = "opni.io/groupKey"
+	NotificationPropertyGroupKey = "opni_group_key"
 	// Opaque identifier for the cluster that generated the notification
-	NotificationPropertyClusterId = "opni.io/clusterId"
+	NotificationPropertyClusterId = "opni_clusterId"
 	// Property that specifies how to classify the notification according to golden signal
-	NotificationPropertyGoldenSignal = "opni.io/goldenSignal"
+	NotificationPropertyGoldenSignal = "opni_goldenSignal"
 	// Property that specifies the severity of the notification. Severity impacts how quickly the
 	// notification is dispatched & repeated, as well as how long to persist it.
-	NotificationPropertySeverity = "opni.io/severity"
+	NotificationPropertySeverity = "opni_severity"
+	// Property that is used to correlate messages to particular incidents
+	NotificationPropertyFingerprint = "opni_fingerprint"
```

Comment:
Out of interest why are we changing the format of labels from the k8s standard label?  Is this something to do with prometheus parsing?

Reply:
yep [prometheus alertmanager labels](https://prometheus.io/docs/alerting/latest/configuration/#configuration-file) don't accept alot of characters ` <labelname>: a string matching the regular expression [a-zA-Z_][a-zA-Z0-9_]* `.